### PR TITLE
Use bash for scheduler

### LINKED ISSUE #47: BQ Data Transfer Auth causes failure in schedule.py
Running the `python3 schedule.py --query_file=changes.sql --table=changes --access_token=${token}` part of `setup.sh` initially fails with the following exception.

```
Traceback (most recent call last):
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/google/api_core/grpc_helpers.py", line 57, in error_remapped_callable
    return callable_(*args, **kwargs)
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/grpc/_channel.py", line 923, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/grpc/_channel.py", line 826, in _end_unary_response_blocking
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INVALID_ARGUMENT
	details = "Failed to find a valid credential. The request to create a transfer config is supposed to contain an authorization code."
	debug_error_string = "{"created":"@1607609612.827897000","description":"Error received from peer ipv4:216.58.207.234:443","file":"src/core/lib/surface/call.cc","file_line":1063,"grpc_message":"Failed to find a valid credential. The request to create a transfer config is supposed to contain an authorization code.","grpc_status":3}"
>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "schedule.py", line 105, in <module>
    app.run(create_or_update_scheduled_query)
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "schedule.py", line 100, in create_or_update_scheduled_query
    response = client.create_transfer_config(parent, transfer_config)
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/google/cloud/bigquery_datatransfer_v1/gapic/data_transfer_service_client.py", line 811, in create_transfer_config
    return self._inner_api_calls["create_transfer_config"](
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/google/api_core/gapic_v1/method.py", line 145, in __call__
    return wrapped_func(*args, **kwargs)
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/google/api_core/retry.py", line 281, in retry_wrapped_func
    return retry_target(
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/google/api_core/retry.py", line 184, in retry_target
    return target()
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/google/api_core/timeout.py", line 214, in func_with_timeout
    return func(*args, **kwargs)
  File "/Users/segan3/Library/Python/3.8/lib/python/site-packages/google/api_core/grpc_helpers.py", line 59, in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
  File "<string>", line 3, in raise_from
google.api_core.exceptions.InvalidArgument: 400 Failed to find a valid credential. The request to create a transfer config is supposed to contain an authorization code.
```

This was from running it with an auth token from my personal account, which had the Owner role in the project. 

There are probably better solutions, but I managed to get around it by first creating a similar scheduled query manually to there get option to grant access to the needed scope.

### LINKED ISSUE #66: Use bash to schedule queries
As seen in #47, using Python to set up the scheduled queries is fragile. I had a better time calling the gcloud CLI directly from bash. Let's replace `/queries/schedule.py` with a bash script.

It's important to preserve the ability to re-run the schedule setter-upper any time (independent of `setup.sh`), so this script should be an independent module that `setup.sh` will invoke.

### PR DESCRIPTION
**Pull request by davidstanke**
fixes #66 (and hopefully #47) -- This PR replaces the Python script for scheduling queries with a bash script that wraps the `bq` CLI. In my testing, it has been more reliable. We need to see if this method works for others.

The `schedule.sh` script can also be run independently of `setup.sh`, by passing a few flags (see `schedule.sh --help` for info).

Additionally, this PR adds support for a bulk cleanup mode to the cleanup script: `cleanup.sh -b` will purge all GCP projects whose ids start with `fourkeys_` or `helloworld_`

### CODE REVIEW

**Comment by dinagraves** [comment]
Awesome!  We need to update here, too: https://github.com/GoogleCloudPlatform/fourkeys#reclassifying-events--updating-your-queries

File: queries/schedule.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+echo QUERY_FILE=$QUERY_FILE
+echo TABLE=$TABLE
+echo PROJECT_ID=$PROJECT_ID
+
+# SCHEDULE THE QUERY
+
+# First, delete the transfer config, if it exists
+for location in US EU; do
+    while [ ! -z "$(bq ls --transfer_config --transfer_location=$location | grep "four_keys_$TABLE" -m 1 | awk '{print $1;}')" ]
```
Comment (dinagraves) [inline queries/schedule.sh:55]: The previous display name was just the table name; eg,`changes` ... `four_keys_changes` is better, but if anyone was running the previous version, it won't find & delete the old `changes` schedules.  WDYT? Break with the past, in with the new?

File: queries/schedule.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+# Usage: 
+# `schedule.sh --query_file=<filename_of_source_query> \
+#  --table=<target_table> --project_id=<GCP_project_containing_dataset>`
+
+help() {
+    printf "Usage: schedule.sh --query_file=<filename_of_source_query> --table=<target_table> --project_id=<GCP_project_containing_dataset>\n"
+}
+
+# PARSE INPUTS
```
Comment (dinagraves) [inline queries/schedule.sh:27]: It wasn't picking up my args, so I simplified the parsing a little bit.

File: setup/cleanup.sh [hunk 3]
```diff
@@ -13,16 +13,47 @@
...
 
 [[ -f "$DIR/env.sh" ]] && echo "Importing environment from $DIR/env.sh..." && . $DIR/env.sh
 
-echo "Deleting projects..."
-set -x
-gcloud projects delete ${FOURKEYS_PROJECT}
-gcloud projects delete ${HELLOWORLD_PROJECT}
-gcloud config set project ${PARENT_PROJECT}
-rm env.sh
+if [[ ${bulk_delete} -gt 0 ]]
```
Comment (dinagraves) [inline setup/cleanup.sh:32]: NICE!  Now if only there was a way to do this with Datastudio reports!

Comment (dinagraves) [reply setup/cleanup.sh:32]: Note: Should document use here https://github.com/GoogleCloudPlatform/fourkeys/blob/main/setup/INSTALL.md#new-google-cloud-projects

**Comment by davidstanke** [comment]
> Awesome! We need to update here, too: https://github.com/GoogleCloudPlatform/fourkeys#reclassifying-events--updating-your-queries

Fixed in 8881e4c

**Review by dinagraves — COMMENTED** [review]
Working on testing it end-to-end!  But can confirm that the individual scheduler worked.  Though I did push a commit on the arg parsing so PTAL!

**Review by dinagraves — APPROVED** [review]
Tested it end-to-end.  Tables populated and scheduled properly.

File: setup/cleanup.sh (hunk 3 above)
Comment (dinagraves) [reply setup/cleanup.sh:32]: I added a note.  Feel free to edit it!

**Comment by dinagraves** [comment]
Do we want to delete the python script?

File: queries/schedule.sh (hunk 2 above)
Comment (davidstanke) [reply queries/schedule.sh:27]: > Do we want to delete the python script?

Ah, yes --thanks! Removed it.

File: queries/schedule.sh (hunk 1 above)
Comment (davidstanke) [reply queries/schedule.sh]: Hmm, you know, if someone were to use this script against an existing project, the behavior would be kind of gross: two scheduled tasks for each table, writing conflicting versions of the derived data.

So I undid that. Thanks for pointing it out!

**Comment by davidstanke** [comment]
Awright, I'm merging. If there are any lingering fixes to do, we can open new issues.

**Pull request merged**